toolchain/lexer/tokenized_buffer.cpp

```diff
@@ -524,13 +524,13 @@ class TokenizedBuffer::Lexer {
   }
 
   auto AddEndOfFileToken() -> void {
-    buffer_.AddToken({.kind = TokenKind::EndOfFile(),
-                      .token_line = current_line_,
-                      .column = current_column_});
+    buffer_->AddToken({.kind = TokenKind::EndOfFile(),
+                       .token_line = current_line_,
+                       .column = current_column_});
   }
 
  private:
-  TokenizedBuffer& buffer_;
+  TokenizedBuffer* buffer_;
```

Comment:
FWIW, especially for members, I would find using the `Nonnull<T*>` worthwhile. It'll give us slightly earlier sanitizer errors if we mess something up and seems like a nice self-documenting thing too.

Clearly a future patch though as we'll want to move the utility.

Reply:
I think this is something to discuss more firmly, to decide whether it's the right style, and if applied it needs to be applied everywhere and consistently. Right now it's not in use in the toolchain.

My concern is that if we use it inconsistently, I think it has a reverse effect of implying places that don't use it are nullable even when they actually aren't. I also think that in the explorer it's been causing friction, and there is a leaning towards removing it because of the friction it creates within C++.

I think this is a case where we should probably pause and figure out whether we're making the right trade-offs before expanding use.